enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1627,8 +1631,11 @@ func (ws *workspace) sync(ctx context.Context) error {
 		return status.InvalidArgumentError("expected at least one of `pushed_branch` or `commit_sha` to be set")
 	}
 
-	if err := ws.fetchRefs(ctx); err != nil {
-		return status.WrapError(err, "fetch refs")
+	if err := ws.fetchPushedRef(ctx); err != nil {
+		return status.WrapError(err, "fetch pushed ref")
+	}
+	if err := ws.fetchTargetRef(ctx); err != nil {
+		return status.WrapError(err, "fetch target ref")
```

Comment:
not 100% sure but in the case where the pushed and target refs are both in the same repo, I think `fetchRefs` would fetch both refs in a single git fetch command. I'd think that might be faster than doing 2 separate commands, since the client and server would have to re-negotiate which blobs need to be fetched?

Reply:
I did a test, and it's about the same (Tested on a couple different branches as a sanity check, and while the results differ, on average the times look similar)

I'd prefer not to pass --merge_with_base as a flag, because we already have so many and it's messy how we have to read the buildbuddy.yaml in both the workflow service and the ci_runner, and the workflow service passes some of the data as a flag, but the ci_runner reads the other half of the data directly from the config

`time git fetch --force origin rb_local_commit master` 
=> 12.99s user 1.81s system 46% cpu 31.689 total
-------
`time git fetch --force origin rb_local_commit ` (30.419s total)
=> 13.16s user 1.87s system 52% cpu 28.750 total
`time git fetch --force origin master `
=> 0.26s user 0.07s system 20% cpu 1.669 total

Reply:
> I'd prefer not to pass --merge_with_base as a flag, because we already have so many and it's messy how we have to read the buildbuddy.yaml in both the workflow service and the ci_runner, and the workflow service passes some of the data as a flag, but the ci_runner reads the other half of the data directly from the config

Sort of a side note (we don't really have to do this now) but another benefit of having this info is that we can skip fetching the `target_branch` when `merge_with_base` is false. I also agree that adding more flags is ugly. Tyler recently added a flag to pass the `Action` as base64, which we could use instead. It's currently only used for Kythe actions, but I think it would be fine to pass it unconditionally

Reply:
D'oh, nevermind - I see that we're fetching the pushed ref first, then reading the config, then deciding whether to fetch the target branch based on that. LGTM